src/Faithlife.Tracing/AspNet/AspNetTracing.cs

```diff
@@ -0,0 +1,91 @@
+#if NET461
+using System;
+using System.Threading;
+using System.Web;
+
+namespace Faithlife.Tracing.AspNet
+{
+	public static class AspNetTracing
```

Comment:
I'd like to see this logic wrapped in a disposable object that could be injected per-request. The static methods could store that object in the HttpContext for clients that don't want to use DI.

```
public sealed class AspNetTracer : IDisposable
{
    public AspNetTracer(HttpRequestBase request) { ... }
    public void BeginTracing() { ... }
    public void EndTracing() { ... }
    public void Dispose() { ... }
}
```

Reply:
> injected per-request

That's too late in the ASP.NET lifecycle.

Reply:
> That's too late in the ASP.NET lifecycle.

Too late for what?